app/routes/adapter-files.js

```diff
@@ -12,13 +12,23 @@ const Express = require('express');
 const MetaScript = require('metascript');
 
 const Router = Express.Router();
+const partnerName = require('../../bin/index');
 
 // =============================================================================
 // MAIN ////////////////////////////////////////////////////////////////////////
 // =============================================================================
 
 const cwd = Process.cwd();
-const fileNames = Fs.readdirSync(cwd);
+var dir = cwd;
+
+if (partnerName !== undefined) {
+    var partnerFolder = partnerName.replace(/\s/g, '').replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
```

Comment:
I'm not sure this works. If we take "Index Exchange" as an example:

1. First the space gets taken away: "IndexExchange".

2. We create a group and look for the first instance of a lowercase letter: we'll find 'n'.

3. We create a group and look for the next instance of an uppercase letter: we'll find 'E'.

4. We replace the original string with the items found in the groups: we end with "i-E".

But what we actually wanted was "index-exchange".

Reply:
Take "AbcDefGhi" for example, /([a-z])([A-Z])/g will match "cD" and "fG". Then replace it following rule '$1-$2'. So "cD" and "fG" will be changed to "c-D" and "f-G". So "AbcDefGhi" will be changed to "Abc-Def-Ghi". After lowercase it will be "abc-def-ghi".